lib/takeaway.rb

```diff
@@ -0,0 +1,60 @@
+require_relative "menu"
+require_relative 'twilio'
+
+
+class Takeaway
+
+  attr_reader :selected_dishes, :total
+
+  def initialize (menu = Menu.new, twilio = Twilio_sms.new)
+    @menu = menu
+    @selected_dishes = Hash.new(0)
+    @total = 0
+    @total_per_dish = 0
+    @twilio = twilio
+  end
+
+  def menu_restaurant
+    @menu.dishes
+  end
+
+  def select_dishes(dish, quantity = 1 )
+    total_order(dish, quantity)
+    @selected_dishes[dish] += quantity
+    info(dish, quantity)
+  end
+
+  def pay(money)
+    if total == money
+      send_message
+      exit
+    else
+      raise "Amount paid not correct"
```

Comment:
Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)